src/WebJobs.Script/Rpc/Configuration/WorkerConfigFactory.cs

```diff
@@ -220,5 +223,37 @@ internal string GetExecutablePathForJava(string defaultExecutablePath)
                 return Path.GetFullPath(Path.Combine(javaHome, "bin", defaultExecutablePath));
             }
         }
+
+        internal string HydrateWorkerPath(string workerPath, string language)
+        {
+            if (string.IsNullOrEmpty(workerPath))
+            {
+                return null;
+            }
+
+            string os = _systemRuntimeInformation.GetOSPlatform().ToString();
+
+            string architecture = _systemRuntimeInformation.GetOSArchitecture().ToString();
+            string version = Environment.GetEnvironmentVariable(LanguageWorkerConstants.FunctionWorkerRuntimeVersionSettingName);
```

Comment:
Add  IEnvironment as dependency and use GetEnvironmentVariable from IEnvironment